app/src/main/java/com/example/sharewhatyoucanproject/donor/AddDonorFragment.kt

```diff
@@ -0,0 +1,200 @@
+                    view: View,
+                    position: Int,
+                    id: Long,
+                ) {
+                    foodTypes[position].also {
+                        addDonorViewModel.foodType = it
+                        requireContext().showToast(it)
+                    }
+                }
+
+                override fun onNothingSelected(parent: AdapterView<*>?) {}
+            }
+
+            binding.uploadbtn.setOnClickListener {
+                addDonorViewModel.title = binding.titleet.text.toString()
+                addDonorViewModel.description = binding.descet.text.toString()
+                addDonorViewModel.hourSet = binding.hourset.text.toString()
+
+                if (!addDonorViewModel.isDataCompleted()) {
+                    requireContext().showToast("Please Fill All Fields")
+                    return@setOnClickListener
+                }
+
+                if (addDonorViewModel.isImageNotSelected()) {
+                    requireContext().showToast("Choose a image")
+                    return@setOnClickListener
+                }
+
+                if (!hasPermissions(activity?.applicationContext, *locationPermissions)) {
+                    requireContext().showToast("Please Allow Location To Post")
+                    activity?.let {
+                        ActivityCompat.requestPermissions(
+                            it,
+                            locationPermissions,
+                            PERMISSION_ALL,
+                        )
+                    }
+                    return@setOnClickListener
+                }
+
+                if (!isGPSEnabled(requireActivity())) {
+                    requireContext().showToast("Open Your GPS")
+                    return@setOnClickListener
+                }
+
+                circularProgressIndicator.visibility = View.VISIBLE
+                if (addDonorViewModel.isFoodNotValid()) {
+                    circularProgressIndicator.visibility = View.GONE
+                    requireContext().showToast("You Cannot Upload Food")
+                    return@setOnClickListener
+                }
+
+                if (ActivityCompat.checkSelfPermission(
+                        requireActivity(),
+                        Manifest.permission.ACCESS_FINE_LOCATION,
+                    ) != PackageManager.PERMISSION_GRANTED && ActivityCompat.checkSelfPermission(
+                        requireActivity(),
+                        Manifest.permission.ACCESS_COARSE_LOCATION,
+                    ) != PackageManager.PERMISSION_GRANTED
+                ) {
+                    // TODO: Consider calling ActivityCompat#requestPermissions
+                }
+                getCurrentLocation()
+            }
+
+            return binding.root
+        }
+
+        @RequiresPermission(anyOf = [Manifest.permission.ACCESS_COARSE_LOCATION, Manifest.permission.ACCESS_FINE_LOCATION])
+        fun getCurrentLocation() {
```

Comment:
before this was in the view model and used the application context, was there an issue? Why move it back here?

Reply:
Moved it to common utils. In part of code refactoring i have moved it.. --FIXED NOW